### PR TITLE
docs: Clarify the approach to release candidates

### PR DESCRIPTION
**Pull request by tico24**
Addresses a conversation in the community meeting held on 19th September 2023.

### Motivation

Clarifies how Release Candidates are handled

### Modifications

Updated documentation

### Verification

`make docs`

### CODE REVIEW

File: docs/releases.md [hunk 1]
```diff
@@ -28,8 +28,11 @@ Both the `argo-server` and `argocli` should be the same version as the controlle
 
 ## Release Cycle
 
-New minor versions are released roughly every 6 months. Release candidates for each major release are typically available
-for 4-6 weeks before the release becomes generally available.
+New minor versions are released roughly every 6 months.
+
+Release candidates (RCs) for each major release are typically available for 4-6 weeks before the release becomes generally available (GA). Features may be shipped in subsequent release candidates.
```
[the remark this answers is not in the document]
Comment (caelan-io) [reply docs/releases.md:33]: Yes, unfortunately, the docs said "Release candidates for major releases..." before. I should have clarified that with @tico24  

We can update this to say, "Release candidates for major and minor releases..." as discussed in the last contributor meeting.

**Review by caelan-io — APPROVED** [review]

**Review by terrytangyuan — APPROVED** [review]

File: docs/releases.md [hunk 2]
```diff
@@ -28,8 +28,11 @@ Both the `argo-server` and `argocli` should be the same version as the controlle
 
 ## Release Cycle
 
-New minor versions are released roughly every 6 months. Release candidates for each major release are typically available
-for 4-6 weeks before the release becomes generally available.
+New minor versions are released roughly every 6 months.
```
Comment (juliev0) [inline docs/releases.md:31]: we're not going to try do quarterly?

Comment (juliev0) [reply docs/releases.md:31]: Also, I assume that if there's backwards incompatibility then we alternatively release a major release but at the same cadence?

File: docs/releases.md [hunk 3]
```diff
@@ -28,8 +28,11 @@ Both the `argo-server` and `argocli` should be the same version as the controlle
...
+
+Release candidates (RCs) for major and minor releases are typically available for 4-6 weeks before the release becomes generally available (GA). Features may be shipped in subsequent release candidates.
+
+When features are shipped in a new release candidate, the most recent release candidate will be available for at least 2 weeks to ensure it is tested sufficiently before it is pushed to GA. If bugs are found with a feature and are not resolved within the 2 week period, the features will be rolled back so as to be saved for the next major release timeline, and a new release candidate will be cut for testing before pushing to GA.
```
Comment (juliev0) [inline docs/releases.md:35]: "for the next major release timeline"? should that be "for the next major/minor release timeline"?

File: docs/releases.md (hunk 2 above)
Comment (juliev0) [reply docs/releases.md:31]: @terrytangyuan what is the cadence you want to advertise? 6 months or 3, or maybe 3-6?

Comment (terrytangyuan) [reply docs/releases.md:31]: lower expectation should be good. I don't think we have resources to do quarterly

Comment (agilgur5) [reply docs/releases.md:31]: As Caelan mentioned, it was adjusted in #11239. This PR does not change that line

Comment (juliev0) [reply docs/releases.md:31]: > As Caelan mentioned, it was adjusted in #11239. This PR does not change that line

Sure

File: docs/releases.md (hunk 3 above)
Comment (caelan-io) [reply docs/releases.md:35]: @tico24 - could you please add this minor clarification to your PR and then we can merge it?

Comment (caelan-io) [reply docs/releases.md]: Thanks! @juliev0 - your feedback was incorporated. Can you please approve and merge for us so we can close this PR out?

Comment (tico24) [reply docs/releases.md]: I also have no reason why the CI failed on a one-word change.

**Review by terrytangyuan — APPROVED** [review]

File: docs/releases.md (hunk 2 above)
Comment (juliev0) [reply docs/releases.md:31]: @tico24 So, when I wrote above "Also, I assume that if there's backwards incompatibility then we alternatively release a major release but at the same cadence?" - I was thinking that maybe we should change the sentence "New minor versions are released roughly every 6 months"? It doesn't say anything about major releases.

**Pull request merged**